---
title: "Capsella rubella introgression to Capsella bursa-pastoris populations"
author: "Maya Wilson Brown"
date: "2024-02-20"
output: html_document
---
```{r}
library(ggplot2)
library(ggrepel)
library(tidyr)
library(stringr)
library(dplyr)
library(rnaturalearth) # for maps
library(scico) # colors
```

I have written functions for the plots I want and selecting K.
We can source the functions from capsella_pop_introgression_functions.R
```{r load functions}
source("~/Documents/PhD/Research/capsella_population_structure/capsella_pop_def_functions.R")
```

```{r directory variables}
# load data directory
admix_dir <- "~/Documents/PhD/Research/capsella_population_structure/new_admix_dat/"
temp_plot <- "~/Documents/PhD/Research/capsella_population_structure/temp_plots/"
plot_dir <- "~/Documents/PhD/Research/capsella_population_structure/plots/"
```

```{r capsella sequenced information}
# load information about samples
vcf_dat <- read.csv("~/Documents/PhD/Research/capsella_sample_info/generated_mkwb/Capsella_vcf_metadata.txt", sep = "\t", header = T)
```

# Capsella bursa-pastoris Principle Components -- All Populations
```{r pca eigenstuff}
# read in eigenvec and eigenval data
vec <- read.csv(paste0(admix_dir, "all_pruned_cbp.eigenvec"), sep = "\t", header = T) #values on each PC

val <-  read.csv(paste0(admix_dir, "all_pruned_cbp.eigenval"), sep = "\t", header = F) #PC variances explained

# percentage of variance explained by each PC
percentage <- round((val/(sum(val))*100), 2)
percentage <- as.matrix(percentage)
percentage <- paste0(names(percentage), " (", percentage, "%)")
```

```{r join pca with meta}
# match individual IDs to informative data
pc_info <- left_join(vec, vcf_dat, join_by("X.IID" == "vcf_sample_name"))

# make continent column, just for colors
# Biologically, not that important because I am arbitrarily assigning a geographic region based on the country name
pc_info <- pc_info %>% mutate(region = case_when(country %in% c("Switzerland","Iceland","Netherlands","Poland","Germany","Sweden","United Kingdom","France") ~ "N. Europe",
                                      country %in% c("Spain", "Greece", "Italy","Jordan", "Algeria","Turkey") ~ "MENA/Mediteranean",
                                      country %in% c("Kyrgyzstan", "Russia") ~ "C. Asia",
                                      country %in% c("China", "Taiwan") ~ "E. Asia",
                                      country == "United States" ~ "US"))
```

```{r pc1_pc2}
pc_plot1 <- ggplot() +
  geom_point(data=pc_info, aes(x=PC1, y=PC2, color = region), size = 3,
             alpha = 0.7) +
  geom_text_repel(data=subset(pc_info, sample_name %in% c("wt-msu", "KYRG-3-14", "WAC5", "HRB132", "FY5", "BEJ3", "TACH1", "TACH_CHIN14")), aes(x=PC1, y=PC2, label = sample_name), size = 2) +
  scale_color_manual(values = region.cols) +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

pc_plot1
```

```{r}
# save plot
ggsave(paste0(plot_dir, "all_cbp_pca1.png"), pc_plot1, height = 5, width = 8)
```

Alternate version for Emily's talk with NYC in pink
```{r alt pc1_pc2}
pc_plot1_alt <- ggplot() +
  geom_point(data=subset(pc_info, citation != "R.Panko"), aes(x=PC1, y=PC2, color = region), size = 3,
             alpha = 0.7) +
  scale_color_manual(values = region.cols) +
  geom_point(data=subset(pc_info, citation == "R.Panko"), aes(x=PC1, y=PC2), size = 3,
             alpha = 0.7, color = "#FF61C3") +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

pc_plot1_alt
```

```{r}
# save plot
ggsave(paste0(plot_dir  , "all_cbp_pca1_pinkNYC.png"), pc_plot1_alt, height = 5, width = 8)
```


Check out the other PCs
```{r pc1_pc3}
# rough plot
pc_plot2  <- ggplot() +
  geom_point(data=pc_info, aes(x=PC1, y=PC3, color = region), size = 3) +
  geom_text_repel(data=subset(pc_info, sample_name %in% c("wt-msu", "KYRG-3-14", "WAC5", "HRB132", "FY5", "BEJ3", "TACH1", "TACH_CHIN14")), aes(x=PC1, y=PC3, label = sample_name), size = 2) +
  scale_color_manual(values = region.cols) +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC3",percentage[3]))


#plot it
pc_plot2
```

```{r}
# save plot
ggsave(paste0(plot_dir, "all_cbp_pca2.png"), pc_plot2, height = 5, width = 8)
```

It seems like PC1 largely captures the US and Northern Europe population separating out from everyone else, and PC2 is picking up on the Mediterranean populations from the larger East Asian group.

Now let's look within NY.
```{r within nyc pca eigenstuff}
# read in eigenvec and eigenval data
vec <- read.csv(paste0(admix_dir, "nyc_pruned_cbp.eigenvec"), sep = "\t", header = T) #values on each PC

val <-  read.csv(paste0(admix_dir, "nyc_pruned_cbp.eigenval"), sep = "\t", header = F) #PC variances explained


# percantage of variance explained by each PC
percentage <- round((val/(sum(val))*100), 2)
percentage <- as.matrix(percentage)
percentage <- paste0(names(percentage), " (", percentage, "%)")
```

```{r join pca with meta}
# match individual IDs to informative data
pc_info <- left_join(vec, vcf_dat, join_by("X.IID" == "vcf_sample_name"))
```

```{r}
# rough plot
nyc_pc_plot1 <- ggplot() +
  geom_point(data=pc_info, aes(x=PC1, y=PC2, color = city)) +
  geom_text_repel(data= subset(pc_info, city %in% c("New York", "Brooklyn", "Queens", "Flushing (Queens)")), aes(x=PC1, y=PC2, label = population), size = 3) +
  theme_classic() +
  scale_color_scico_d(palette = "managua") +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

nyc_pc_plot1
```

Save image
```{r}
# save plot
ggsave(paste0(plot_dir, "nyc_cbp_pca1.png"), nyc_pc_plot1, height = 3, width = 6)
```

For NYC, PC1 is capturing variation driven by population 24 (New Jersey)

Taking a look at PC3 as well.
```{r}
# rough plot
nyc_pc_plot2 <- ggplot() +
  geom_point(data=pc_info, aes(x=PC2, y=PC3, color = city)) +
  geom_text_repel(data= subset(pc_info, city %in% c("New York", "Brooklyn", "Queens", "Flushing (Queens)")), aes(x=PC2, y=PC3, label = population), size = 3) +
  theme_classic() +
  scale_color_scico_d(palette = "managua") +
  labs(x=paste0("PC2",percentage[2]), y= paste0("PC3",percentage[3]))

nyc_pc_plot2
```

Is there spatial structure to PC2 or PC3 in NYC?
```{r}
ny_map + geom_point(data=pc_info, aes(x=longitude, y=latitude, color = PC2), size = 3, alpha = 0.8)
```


```{r}
# rough plot
nyc_pc_plot2 <- ggplot() +
  geom_point(data=pc_info, aes(x=PC1, y=PC3, color = city)) +
  geom_text_repel(data= subset(pc_info, city %in% c("New York", "Brooklyn", "Queens", "Flushing (Queens)")), aes(x=PC1, y=PC3, label = population), size = 3) +
  theme_classic() +
  scale_color_scico_d(palette = "managua") +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC3",percentage[3]))

nyc_pc_plot2
```


#### ADMIXTURE conditioned on K

If we just think about ADMIXTURE and STRUCTURE as ways to select a valuable K number of ancestral groups, it makes sense to run admixture on the Eurasian *C. bursa-pastoris* samples on their own, and select the best K groups from the ancestral population. Then, using that K, I would run ADMIXTURE on only the NYC populations and assign the NYC population to those K ancestral groups.

```{r import admixture output files}
# read in the PLINK fam file which has the sample name information in the same order as the admixture output files
ea_fam <- read.delim(paste0(admix_dir, "eurasian_pruned_cbp.fam"), header=FALSE)

# list the files in admixture directory matching suffix pattern
lst <- list.files(path = admix_dir, pattern = "eurasian_pruned_cbp.[0-9]+.Q") # Q files are ancestry fractions for each sample

ea_ancestry_propsK <- list()  # Create an empty list to store the data for each iteration of K groups
#loop to load sample data
for (file in lst) {
  file_path <- file.path(admix_dir, file)  # Get the full file path
  ea_ancestry_propsK[[file]] <- read.table(file_path, header = F)  # Read the file and store in the list
}
```

```{r}
# create sample name key from fam file 
ea_name_key <- left_join(data.frame(ea_fam$V2), vcf_dat[,c("vcf_sample_name", "sample_name", "country")],
                         join_by("ea_fam.V2" == "vcf_sample_name"))
colnames(ea_name_key)[1] <- "vcf_sample_name"

# add region to key
ea_name_key <- ea_name_key %>% mutate(region = case_when(country %in% c("Switzerland","Iceland","Netherlands","Poland","Germany","Sweden","United Kingdom","France") ~ "N. Europe",
                                      country %in% c("Spain", "Greece", "Italy","Jordan", "Algeria","Turkey") ~ "MENA/Mediterranean",
                                      country %in% c("Kyrgyzstan", "Russia") ~ "C. Asia",
                                      country %in% c("China", "Taiwan") ~ "E. Asia",
                                      country == "United States" ~ "US"))



# bind fam file to all ADMIXTURE files
#ea_p <- lapply(ea_ancestry_propsK, transform, vcf_sample_name=ea_fam$V2)
ea_p <- Map(cbind, ea_ancestry_propsK, list(ea_name_key))

# name list elements by the K groups they represent
names(ea_p) <- paste0("K",str_extract(names(ea_p), "[0-9]+"))
rm(ea_ancestry_propsK)
```

```{r cv_error K groups}
# read in Cross Validation (cv) error log
ea_cv <- read.delim(paste0(admix_dir, "eua_cv_error.log"), header = FALSE, sep = ":")
# extract K groups from string
ea_cv$K_groups <- str_extract(ea_cv$V1, "[0-9]+")

# Plot
plotK(ea_cv)

# select K
#K = selectK(ea_cv)
```

```{r}
# data for k = 2 on Eurasian group
ea_k3 <- ancestry_cols(ea_p, "K3", ggplot_opt = scale_fill_manual(values= c("pop2" = "#B2F2FD", "pop3"= "#996330", "pop1"= "#7FC55F"),
                                                                  labels = c("pop2" = "E. Asia",
                                                                             "pop3" = "MENA/Mediterranean",
                                                                             "pop1" = "N. Eurasia")))
# make just the plot show up
plot(ea_k3[[1]])

```

```{r}
ggsave(paste0(plot_dir,"eurasian_admix_k3_legend_bottom.png"), ea_k3[[1]], width = 13, height = 5, units = "in")
```

```{r, eval = F, echo=FALSE}
# data for k = 2 on Eurasian group
ea_k4 <- ancestry_cols(ea_p, "K4")
# make just the plot show up
plot(ea_k4[[1]])
```

```{r, eval=FALSE, echo=FALSE}
ggsave(paste0(temp_plot,"eurasian_cbp_k5.png"), ea_k5[[1]], width = 13, height = 5, units = "in")
```

Same information largely, but on a map.

```{r assign ancestry}
# assign sample to majority ancestry
ea_k3_ancestry <- assign_ancestry(ea_p, "K3")

# join with location data
ea_k3_ancestryL <- left_join(ea_k3_ancestry, vcf_dat[c("vcf_sample_name","sample_name",
                                                 "longitude","latitude","latlong_approx")])
```

```{r}
# world map
ea_k3_map <- eurasia + geom_point(data=ea_k3_ancestryL, aes(x=longitude, y=latitude, color = group, shape = region), size = 7, alpha = 0.8) +
  scale_color_manual(values= c("pop2" = "#99D9E4", "pop3"= "#996330", "pop1"= "#7FC55F"),
                    labels = c("pop2"= "E. Asia", pop3 = "MENA/Mediterranean", "pop1" = "N. Eurasia")) + scale_shape_manual(values = c('C. Asia'=15, 'E. Asia'=18, 'N. Europe'=16, 'MENA/Mediterranean'=17))

ea_k3_map
```

```{r}
ggsave(paste0(plot_dir, "eurasian_k3_map.png"),ea_k3_map, width = 13, height = 7, units = "in")
```


```{r}
# assign sample to majority ancestry
ea_k4_ancestry <- assign_ancestry(ea_p, "K4")

# join with location data
ea_k4_ancestryL <- left_join(ea_k4_ancestry, vcf_dat[c("vcf_sample_name","sample_name",
                                                 "longitude","latitude","latlong_approx")])
```

```{r}
ea_k4_map <- eurasia + geom_point(data=ea_k4_ancestryL, aes(x=longitude, y=latitude, color = group, shape = region), size = 7, alpha = 0.8) +
  scale_shape_manual(values = c('C. Asia'=15, 'E. Asia'=18, 'N. Europe'=16, 'MENA/Mediterranean'=17)) +
  geom_text_repel(data=subset(ea_k4_ancestryL, sample_name %in% c("RK32", "KYRG-3-14","QD325","wt-msu", "BEJ3", "FY5", "TACH1", "TACH-CHIN14", "HRB138", "HRB135", "HRB132", "QD325", "HD63")), aes(x=longitude, y=latitude, label = sample_name), max.overlaps = 50, size = 3)

ea_k4_map
```


```{r}
ggsave(paste0(temp_plot, "eurasian_k5_map_labels.png"),ea_k5_map, width = 13, height = 7, units = "in")
```

```{r}
ea_k4_map <- eurasia + geom_point(data=ea_k4_ancestryL, aes(x=longitude, y=latitude, color = group, shape = region), size = 7, alpha = 0.8) +
  scale_shape_manual(values = c('C. Asia'=15, 'E. Asia'=18, 'N. Europe'=16, 'MENA/Mediterranean'=17)) +
  geom_text_repel(data=subset(ea_k4_ancestryL, sample_name %in% c("AQ415","HJC419","JJN2","LJH9","YNHZ2","DL174","LJ5","LL761","NE2")), aes(x=longitude, y=latitude, label = sample_name), max.overlaps = 50, size = 3)

ea_k4_map
```


### NYC Metro population structure conditional on Eurasian allele frequency
```{r nyc import admixture output files}
# read in the PLINK fam file which has the sample name information in the same order as the admixture output files
nyc_fam <- read.delim(paste0(admix_dir, "nyc_pruned_cbp.fam"), header=FALSE)

# list the files in admixture directory matching suffix pattern
lst <- list.files(path = admix_dir, pattern = "nyc_pruned_cbp.[0-9]+.Q") # Q files are ancestry fractions for each sample

nyc_ancestry_propsK <- list()  # Create an empty list to store the data for each iteration of K groups
#loop to load sample data
for (file in lst) {
  file_path <- file.path(admix_dir, file)  # Get the full file path
  nyc_ancestry_propsK[[file]] <- read.table(file_path, header = F)  # Read the file and store in the list
}
```

```{r}
# create sample name key from fam file 
nyc_name_key <- left_join(data.frame(nyc_fam$V2), vcf_dat[,c("vcf_sample_name", "sample_name", "country")],
                         join_by("nyc_fam.V2" == "vcf_sample_name"))
colnames(nyc_name_key)[1] <- "vcf_sample_name"
# add region to key
nyc_name_key <- nyc_name_key %>% mutate(region = case_when(country %in% c("Switzerland","Iceland","Netherlands","Poland","Germany","Sweden","United Kingdom","France") ~ "N. Europe",
                                      country %in% c("Spain", "Greece", "Italy","Jordan", "Algeria","Turkey") ~ "MENA/Mediterranean",
                                      country %in% c("Kyrgyzstan", "Russia") ~ "C. Asia",
                                      country %in% c("China", "Taiwan") ~ "E. Asia",
                                      country == "United States" ~ "US"))

# bind fam file to all ADMIXTURE files
#nyc_p <- lapply(nyc_ancestry_propsK, transform, vcf_sample_name=nyc_fam$V2)
nyc_p <- Map(cbind, nyc_ancestry_propsK, list(nyc_name_key))

# name list elements by the K groups they represent
names(nyc_p) <- paste0("K",str_extract(names(nyc_p), "[0-9]+"))
rm(nyc_ancestry_propsK)
```


```{r}
# data for k = 2 on Eurasian group
nyc_k3 <- ancestry_cols(nyc_p, "K3",ggplot_opt = scale_fill_manual(values= c("pop2" = "#B2F2FD", "pop3"= "#996330", "pop1"= "#7FC55F"),
                                                                   labels = c("pop2" = "E. Asia",
                                                                             "pop3" = "MENA/Mediterranean",
                                                                             "pop1" = "N. Eurasia")))
# make just the plot show up
plot(nyc_k3[[1]])
```

```{r}
ggsave(paste0(plot_dir,"nyc_cbp_k3.png"), nyc_k3[[1]], width = 13, height = 5, units = "in")
```

Prep data for NYC map
```{r}
# assign sample to majority ancestry
nyc_k3_ancestry <- assign_ancestry(nyc_p, "K3")

# join with location data
nyc_k3_ancestryL <- left_join(nyc_k3_ancestry, vcf_dat[c("vcf_sample_name","sample_name",
                                                 "longitude","latitude","latlong_approx", "city")])
```

Just a map colored by population for explaining the dataset
```{r}
nyc_pops_map <- ny_map + geom_point(data=nyc_k3_ancestryL, aes(x=longitude, y=latitude, color = city), size = 3, alpha = 0.8) + scale_color_scico_d(palette = "managua")

```

```{r}
ggsave(paste0(plot_dir,"nyc_pops_map.png"),nyc_pops_map, width = 7, height = 3, units = "in")
```


Since the mapped plot of only NYC will not be that interesting, I will combine the Eurasian and NYC data frames to make a map.
```{r}
# combine Eurasian and NYC data frames
k3_ancestry <- rbind(ea_k3_ancestry, nyc_k3_ancestry)

# join with location data
k3_ancestryL <- left_join(k3_ancestry, vcf_dat[c("vcf_sample_name","sample_name",
                                                 "longitude","latitude","latlong_approx")])
```

```{r}
cond_k3_map <- world + geom_point(data=k3_ancestryL, aes(x=longitude, y=latitude, fill = group, color = latlong_approx), shape=21, size = 7) 

cond_k3_map
```

```{r}
ggsave(paste0(temp_plot, "cond_k3_bigDots.png"),cond_k3_map, width = 13, height = 7, units = "in")
```

With all of the above plots, the East Asian population always stands out. There is evidence that there is more continuous structure or some sort of gradient moving from the Mediterranean region, through central Asia (Kyrgyzstan).

This is also good evidence that we can use some individuals from the East Asian group as a parental population in the local ancestry inference.

~~Regardless of whether K=2,3, or 4, there are always 33 individuals that are in the East Asian cluster with no admixture present. I think it makes sense to use 25-30 individuals as the parental group in the LAI, and verify there's no inferred *C. rubella* ancestry with the rest.~~

# NYC and European with/out MENA 
Seeing as the NYC individuals seems to clearly have a lot of shared variation with the Northern European group, it is worth seeing how NYC individuals partition into this group.

The N. Eurasian cluster *without NYC included* is only 22 samples. PLINK will not perform PCA with less than 50 samples without being given an allele frequency file because the sample number is too low. I think I should still be able to convert the pruned VCF to bed files and do the conditional ADMIXTURE analysis.

I do not know if that will give me any additional useful information, however. Given how the PCA looks, it seems pretty clear that the NYC samples are most similar to Polish, Russian and Swedish populations, with the New Jersey population being the closest. 

Also, Northern Chinese populations cluster together, while the sample from northeastern China clusters with the sample from Reykjavik.

**NOTE:** Originally, I included Mediterranean/Middle Eastern,North African individuals in the analysis, but it became clear that was not useful or informative since they form their own cluster quite removed from the N. Europe and NYC individuals.

For the following analyses, I only include samples that have majority N. European ancestry when K=3 in the ancestral range with individuals from New York and New Jersey.

Plan:
- PCA with NYC + N. Europe
- Admixture free on NYC + N.EU (to see if the CV error rates suggest NYC wants to cluster on its own)
  - if NYC does cluster on its own freely with N.EU, that *could* suggest there's been quite a bit of allele frequency change across continents **OR** the NYC group is just large and it popping out as its own cluster could be an artifact of over fitting (? but I really do not know if that matters and would have to look at the documentation for ADMIXTURE)

```{r select Euro group}
# select individuals whose ancestry is not the East Asian group: pop1 when k=3
nEU <- ea_k3_ancestry[which(ea_k3_ancestryL$group == "pop1"),]

# remove samples drawn out by PC3 in the whole CBP PCA
nEU <- nEU[which(!nEU$sample_name %in% c("wt-msu", "BEJ3", "FY5","TACH1", "KYRG-3-14")),]
```

```{r}
#save and write files
write.table(nEU$vcf_sample_name, file = "~/Documents/PhD/Research/capsella_population_structure/vcf_neu.txt", quote = F, row.names = F, col.names = F, sep = "\t")
```


I would also like to do a PCA with the East Asian individuals just to see how they compare. I will not have enough individuals for PLINK to do the PCA, so I will have to make an allele frequency file.
```{r select EAsia group}
# select individuals whose ancestry is not the East Asian group: pop1 when k=3
easia <- ea_k3_ancestry[which(ea_k3_ancestryL$group == "pop2"),]
```

```{r}
#save and write files
write.table(easia$vcf_sample_name, file = "~/Documents/PhD/Research/capsella_population_structure/vcf_easia.txt", quote = F, row.names = F, col.names = F, sep = "\t")
```

# Northern European & NYC Plots

Starting with PCA of NYC with N. European group (minus Moscow and some China).

```{r nyc and n europe pca eigenstuff}
# read in eigenvec and eigenval data
vec <- read.csv(paste0(admix_dir, "nyc_neu.eigenvec"), sep = "\t", header = T) #values on each PC

val <-  read.csv(paste0(admix_dir, "nyc_neu.eigenval"), sep = "\t", header = F) #PC variances explained


# percantage of variance explained by each PC
percentage <- round((val/(sum(val))*100), 2)
percentage <- as.matrix(percentage)
percentage <- paste0(names(percentage), " (", percentage, "%)")
```

```{r join pca with meta}
# match individual IDs to informative data
pc_info <- left_join(vec, vcf_dat, join_by("X.IID" == "vcf_sample_name"))
```

```{r}
# rough plot
nyc_neu_pc_plot1 <- ggplot() +
  geom_point(data=pc_info, aes(x=PC1, y=PC2, color = country), size = 3, alpha = 0.75) +
  scale_color_discrete(labels = c("United States" = "NYC Metro")) +
  geom_text_repel(data=subset(pc_info, country %in% c("China")), aes(x=PC1, y=PC2, label = sample_name), max.overlaps = 50, size = 2) +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

nyc_neu_pc_plot1
```
PC1 seems to separate some of the Chinese plants from the rest, while PC2 clusters mainland Europe from others.

Save image
```{r}
# save plot
ggsave(paste0(plot_dir, "nyc_neu_cbp_pca1.png"), nyc_neu_pc_plot1, height = 5, width = 8)
```

I would like to zoom in on that cluster
```{r}
nyc_neu_pc_plot1_zoom <- ggplot() +
  geom_point(data=subset(pc_info, country %in% c("United States", "Netherlands", "Poland", "Russia", "Sweden")), aes(x=PC1, y=PC2, color = country),
             size = 3, alpha = 0.75) +
  scale_color_discrete(labels = c("United States" = "NYC Metro")) +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

nyc_neu_pc_plot1_zoom
```

```{r}
ggsave(paste0(plot_dir, "nyc_neu_pca1_zoom.png"), nyc_neu_pc_plot1_zoom, height = 5, width = 8)
```
Looks East to West (PC1) and North to South (PC2)?

We can check that
```{r}
nyc_neu_pc1Long_zoom <- ggplot() +
  geom_point(data=subset(pc_info, country %in% c("United States", "Netherlands", "Poland", "Russia", "Sweden")), aes(x=longitude, y=PC1, color = country)) +
  theme_bw() +
  labs(x="longitude", y= paste0("PC1",percentage[1])) +
  geom_text_repel(data=subset(pc_info, country %in% c("United States", "Netherlands", "Poland", "Russia", "Sweden")), aes(x=longitude, y=PC1, label = sample_name), max.overlaps = 50, size = 2)

nyc_neu_pc1Long_zoom
```

```{r}
# save plot
ggsave(paste0(temp_plot, "nyc_neu_pca1Long_zoom.png"), nyc_neu_pc1Long_zoom, height = 5, width = 8)
```

```{r}
nyc_neu_pc1Lat_zoom <- ggplot() +
  geom_point(data=subset(pc_info, country %in% c("United States", "Netherlands", "Poland", "Russia", "Sweden")), aes(x=PC2, y=latitude, color = country)) +
  theme_bw() +
  labs(x=paste0("PC2",percentage[2]), y= "latitude") +
  geom_text_repel(data=subset(pc_info, country %in% c("United States", "Netherlands", "Poland", "Russia", "Sweden")), aes(x=PC2, y=latitude, label = sample_name), max.overlaps = 50, size = 2)

nyc_neu_pc1Lat_zoom
```

```{r}
# save plot
ggsave(paste0(temp_plot, "nyc_neu_pca1Lat_zoom.png"), nyc_neu_pc1Lat_zoom, height = 5, width = 8)
```

Yeah, I might say so.

The N. European cluster without NYC included is only 22 samples. PLINK will perform PCA with less than 50 samples without being given an allele frequency file because the sample number is too low. I think I should still be able to convert the pruned VCF to bed files and do the conditional ADMIXTURE analysis.

I do not know if that will give me any additional useful information, however. Given how the PCA looks, it seems pretty clear that the NYC samples are most similar to Polish, Russian and Swedish populations, with the New Jersey population being the closest. 

Also, Northern Chinese populations cluster together, while the sample from northeastern China clusters with the sample from Reykjavik.

# East Asian Samples
```{r e asia pca eigenstuff}
# read in eigenvec and eigenval data
vec <- read.csv(paste0(admix_dir, "easia_cbp.eigenvec"), sep = "\t", header = T) #values on each PC

val <-  read.csv(paste0(admix_dir, "easia_cbp.eigenval"), sep = "\t", header = F) #PC variances explained


# percantage of variance explained by each PC
percentage <- round((val/(sum(val))*100), 2)
percentage <- as.matrix(percentage)
percentage <- paste0(names(percentage), " (", percentage, "%)")
```

```{r}
# match individual IDs to informative data
pc_info <- left_join(vec, vcf_dat, join_by("X.IID" == "vcf_sample_name"))
```

```{r}
# rough plot
easia_pc_plot1 <- ggplot() +
  geom_point(data=pc_info, aes(x=PC1, y=PC2), size = 3, alpha = 0.75) +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

easia_pc_plot1
```
I kind of want to color it by K=4 colors so
```{r}
pc_info_col <- left_join(pc_info, ea_k4_ancestryL)
```
```{r}
# rough plot
easia_pc_plot1 <- ggplot() +
  geom_point(data=pc_info_col, aes(x=PC1, y=PC2, color = V2), size = 3, alpha = 0.75) +
  geom_text_repel(data=subset(pc_info_col, sample_name %in% c("AQ415","HJC419","JJN2","LJH9","YNHZ2","DL174","LJ5","LL761","NE2")), aes(x=PC1, y=PC2, label = sample_name), max.overlaps = 50, size = 2) +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

easia_pc_plot1
```
```{r}
# rough plot
easia_pc_plot1 <- ggplot() +
  geom_point(data=pc_info_col, aes(x=PC1, y=PC2, color = V2, size = latitude), alpha = 0.75) +
  geom_text_repel(data=subset(pc_info_col, sample_name %in% c("AQ415","HJC419","JJN2","LJH9","YNHZ2","DL174","LJ5","LL761","NE2")), aes(x=PC1, y=PC2, label = sample_name), max.overlaps = 50, size = 2) +
  scale_color_gradient(high = "green", low = "blue") +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC2",percentage[2]))

easia_pc_plot1
```

```{r}
# rough plot
easia_pc_plot2 <- ggplot() +
  geom_point(data=pc_info_col, aes(x=PC1, y=PC3, color = V2), size = 3, alpha = 0.75) +
  geom_text_repel(data=subset(pc_info_col, sample_name %in% c("AQ415","HJC419","JJN2","LJH9","YNHZ2","DL174","LJ5","LL761","NE2")), aes(x=PC1, y=PC3, label = sample_name), max.overlaps = 50, size = 2) +
  scale_color_gradient(high = "green", low = "blue") +
  theme_classic() +
  labs(x=paste0("PC1",percentage[1]), y= paste0("PC3",percentage[3]))

easia_pc_plot2
```
```{r}
# I think I am just adding the 3D plot for fun?
library(plotly)

plot_ly(data = pc_info_col, x = pc_info_col$PC1, y=pc_info_col$PC2, z=pc_info_col$PC3,type="scatter3d", mode="markers", color = pc_info_col$longitude)
```


# Write populations to file
At this point, it makes sense to generate a key for generalized population structure to use in other analyses.
Specifically, I am going to use these population groups in the PIXY analyses, and other explorations of comparisons between the populations at the local genomic scale.

Here, I just plan to combine the population keys generated above. I consider the conditional analysis to reflect the truest and most probable population structure.

The most useful and informative clustering is when K=3, so I will use that to define clusters

```{r}
# create descriptive column for ancestral range population groups
k3Pops <- k3_ancestry %>% mutate(k3population = case_when(group == "pop1" ~ "N_Europe",
                                                         group == "pop2" ~ "E_Asia",
                                                         group == "pop3" ~ "MENA"))
```

Now I will write the data population data to file
```{r save cbp population file, eval=FALSE}
write.table(k3Pops[c("vcf_sample_name","sample_name","k3population")], file = "~/Documents/PhD/Research/capsella_population_structure/cbp_pop_str.txt", col.names = T, row.names = F, quote = F, sep = "\t")
```
