---
title: "Admixture anaylsis"
author: "Maya Wilson Brown"
date: "2024-02-26"
output: html_document
---

Admixture analysis of global *Capsella bursa-pastoris* samples.

### Selecting C. bursa-pastoris samples from VCF

First, I will compare all potential samples to samples that made it into the final VCF
```{bash, eval = F, echo = FALSE}
# on HPCC
cd /mnt/research/josephslab/Maya/capsella/vcf/adrian_vcf/
# get sample names from VCF
bcftools query -l 
```

```{r}
library(stringr)
library(dplyr)
```

```{r}
# load vcf metadata
vcf_meta <- read.csv("~/Documents/PhD/Research/capsella_sample_info/generated_mkwb/Capsella_vcf_metadata.txt", sep = "\t", header = T)
```

```{r, eval =FALSE}
# load individuals in the final filtered version of the VCF
invcf <- read.csv("~/Documents/PhD/Research/vcf_filtering/individuals/j19_in_vcf.txt",
                  header = F)

# join the information
vcf_dat <- left_join(invcf, vcf_meta, join_by("V1" == "vcf_sample_name"))
```
```{r, eval=FALSE}
# collect only cbp
cbp <- vcf_dat[vcf_dat$species == "Capsella bursa-pastoris",]
```

```{r all Capsella bursa-pastoris in VCF, eval=FALSE}
#write C. bursa-pastoris VCF names to list
write.table(cbp$V1, "~/Documents/PhD/Research/capsella_population_structure/all_cbp.txt",
            quote = F, sep = "\t", row.names = F, col.names = F)
```

# Conditioning ancestry proportions on clusters in a reference panel -- projection analysis
For this analysis, I wanto to use *Capsella bursa-pastoris* in Eurasia to generate populations in the native range, and assign individuals in the introduced range to those populations.

This requires that the same SNP set be in both datasets. So I will mark SNPs in LD on the whole dataset, and then prune them on the Eurasian samples and NYC samples to generate two sets .bed, .bim, and .fam files

I also need to make a list of NYC and Eurasian samples. I know that WAC5 is the only other sample in North America that is not from NYC (aside from the reference which could be from Reno?? It does not have location data associated, and in the PCA, it stands out quite a lot so I will remove it).

First, to make the data sets that will go into this analysis
```{r, eval=FALSE}
# selecting NYC only samples
nyc <- cbp[which(cbp$citation == "R.Panko"),]

# eurasian samples, also removing WAC5 because it is not in Eurasia
# WAC5 is from Waco, Texas
eurasia <- subset(cbp, !(sample_name %in% c("WAC5", nyc$sample_name)))
```

```{r, eval=FALSE}
# write both datasets to file
write.table(nyc$V1, "~/Documents/PhD/Research/capsella_population_structure/vcf_cbp_nyc.txt", quote = F, sep = "\t", row.names = F, col.names = F)

write.table(eurasia$V1, "~/Documents/PhD/Research/capsella_population_structure/vcf_cbp_eurasia.txt", quote = F, sep = "\t", row.names = F, col.names = F)
```

From the **Projection analysis** section of the [ADMIXTURE 1.3 manual](https://dalexander.github.io/admixture/admixture-manual.pdf):

>ADMIXTURE now allows loading of learned allele frequencies from the .P files. For two
datasets with the same set of SNPs, clusters can be learned using the unsupervised mode
of ADMIXTURE on the first dataset; subsequently, the learned clusters and ancestry
proportions from the first dataset can be provided as input used to project the second
dataset with the -P option.
Example: Suppose reference.bed is the PLINK binary file containing reference panels
and study.bed is the PLINK binary file containing study samples. The following sequence
of commands can be used to learn population structure from the reference panel and project
the study individuals on it:

```{bash, eval = F}
 #Verify the two datasets have the same set of SNPs
 diff -s reference.bim study.bim
 #Run unsupervised ADMIXTURE with K=2
 admixture reference.bed 2
 #Use learned allele frequencies as (fixed) input to next step
 cp reference.2.P study.2.P.in
 #Run projection ADMIXTURE with K=2
 admixture -P study.bed 2
```

>The projection approach is particularly useful when the study sample is much larger than
the reference panel. When the study sample is comparable in size to the reference panel,
it may be preferable to run unsupervised ADMIXTURE on the combined set since the
running time scales linearly with the number of samples.

>The same approach can also be used for inferring ancestry on a set of related individuals.
Pedigree information or existing methods such as PLINK [5], KING [6] or PRIMUS [7] can
be used to identify the largest subset of unrelated individuals in the dataset. ADMIXTURE
is then run unsupervised on the largest set of unrelated individuals (using that as the
reference set in the example above). The rest of the individuals (who can be related to the
reference set) can then be projected on the resulting population structure.

```{bash, code = readLines("~/Documents/PhD/Research/capsella_population_structure/runConditionalAdmixture.sh"), eval = FALSE}
```