---
title: "Admixture anaylsis"
author: "Maya Wilson Brown"
date: "2024-02-26"
output: html_document
---

Admixture analysis of global *Capsella bursa-pastoris* samples.

Update: May 28, 2024; Adrian has new versions of VCFs aligned ot the NYC genome.

### Selecting C. bursa-pastoris samples from VCF

First, I will compare all potential samples to samples that made it into the final VCF
```{bash, eval = F}
# on HPCC
cd /mnt/research/josephslab/Maya/capsella/vcf/adrian_vcf/
# get sample names from VCF
#bcftools query -l CbpCr_Cbp_new_filtered.vcf.gz > Capsella_vcf_samples.txt
bcftools query -l final_called_all.CBPCRCG.v.vcf > cbp_crcg_vcf_sample_names.txt
```

```{r}
library(stringr)
library(dplyr)
```

```{r}
# load whole genome sequence sample data
wgs <- read.csv("~/Documents/PhD/Research/capsella_sample_info/generated_mkwb/Capsella_combined_wgs_v2.txt", sep = "\t", header = T)

# load VCF sample list
#vcf <- read.table("~/Downloads/Capsella_vcf_samples.txt", quote="\"", comment.char="")
vcf <- read.table("~/Desktop/cbp_crcg_vcf_sample_names.txt", quote="\"", comment.char="")
```

```{r}
# match the vcf samples to the whole genome data
# make new column that removes the trailing S from NYC samples
vcf$simplified_name <- str_remove_all(vcf$V1, "_S[0-9]+_L003")
# remove also the .sam from the other sample names
vcf$simplified_name <- str_remove_all(vcf$simplified_name, ".sam")
```

I want to double check that I have metadata for all the samples in the VCF.
```{r}
# rows in VCF that are not in wgs 
# should only be Panko NYC that are missing SRA run accessions)
setdiff(vcf$simplified_name, wgs$run_accession)
```

I want to be able to subset them easily later so I will join the VCF name with the metadata.
```{r}
# join by sample_name (Panko to vcf)
vcf_panko <- left_join(vcf, wgs, join_by("simplified_name" == "sample_name"))
# add sample name back to panko
vcf_panko$sample_name <- vcf_panko$simplified_name

# join by run accession (SRA data to VCF)
vcf_SRA <- left_join(vcf, wgs, join_by("simplified_name" == "run_accession"))
# add run accession back to SRA
vcf_SRA$run_accession <- vcf_SRA$sample_name


```

wt_msu and run samples called "bursa-pastoris" are actually the same
```{r}
# The first occurrence is actually more informative so I will keep that one.
vcf_SRA <- vcf_SRA[!duplicated(vcf_SRA[1:2]), ]
```


```{r}
# joining all the information
full_info <- rows_patch(vcf_panko, vcf_SRA)

#checking if simplified_name is the same as sample_name
sum(full_info$simplified_name == full_info$sample_name) == nrow(full_info)

# cool, I will make simplified name into sample name
full_info <- full_info[-16] #drop last column (sample_name)
colnames(full_info)[which(colnames(full_info) == "simplified_name")] <- "sample_name" #change the same of simplified_name to sample_name
```


VCF information: I only want Capsella bursa-pastoris ones so, I will just select those
```{r}
# collect only cbp
cbp <- full_info[full_info$species == "Capsella bursa-pastoris",]
```

Finally, write to table for use on HPCC.
```{r all Capsella bursa-pastoris in VCF}
#write C. bursa-pastoris VCF names to list
write.table(cbp$V1, "~/Downloads/vcf_cbp_only_0606_2024.txt", quote = F, sep = "\t", row.names = F, col.names = F)
```

I want do redo the admixture analysis with only samples in the European group (which precipitates out of the admixture analysis with all the *Capsella bursa-pastoris* above)
```{r}
# list of individuals who have over 60% ancestry in European group when k=3
eu_cbp <- read.table("~/Downloads/eu_cbp_k3.txt", quote="\"", comment.char="", header = F)
```

### Admixture analysis
Admixture assumed independence of variants, so we need to LD prune the SNPs.
Full Script below

```{bash LD prune admixture}
#!/bin/bash
#
#SBATCH --job-name=LDpruneAdmixture
#SBATCH --mem=10G
#SBATCH --time=5:00:00         #realistically, shouldn't take very long
#SBATCH --mail-type=ALL
#SBATCH --mail-user=wils1582@msu.edu
#SBATCH --cpus-per-task=15
#SBATCH --output=/mnt/scratch/wils1582/slurm/slurm-%A_%a.out

### VARIABLES
VCF=/mnt/research/josephslab/Maya/capsella/vcf/CbpCr_Cbp_new/CbpCr_Cbp_new_filtered.vcf.gz
SAMPLES=/mnt/research/josephslab/Maya/capsella/admixture/vcf_cbp_only.txt
OUTDIR=/mnt/research/josephslab/Maya/capsella/admixture/

# move to directory
cd $OUTDIR

#load modules
module purge
module load PLINK/2.00a_10.2-x86_64
module load ADMIXTURE/1.3.0

#### LD Prune and output BED
# Identify SNPs in LD
# Parameters mark SNPs
plink2 --vcf $VCF \
	--indep-pairwise 100 5 0.2 \
	--keep $SAMPLES \
	--allow-extra-chr \
	--out pruned_snps \
	--set-all-var-ids @:#

# Prune LD SNPs
# Variant IDs set to be 'scaffold:position'
# ADMIXTURE requires the input be in PLINK .bed format, so here I create the bed file as well.
plink2 --vcf $VCF \
  --extract pruned_snps.prune.in \
  --make-bed \
  --out plink2_pruned_cbp \
	--allow-extra-chr \
	--keep $SAMPLES \
	--set-all-var-ids @:#

# ADMIXTURE only takes chromosomes as numbers so in all the PLINK output files, SCF_ needs to be removed
sed -i 's/^SCF_//g' plink2_pruned_cbp.bed
sed -i 's/^SCF_//g' plink2_pruned_cbp.bim
sed -i 's/^SCF_//g' plink2_pruned_cbp.fam

####### ADMIXTURE
# move to directory (double check)
cd /mnt/research/josephslab/Maya/capsella/admixture

# admixture run with cross validation
for K in 1 2 3 4 5 6; \
	do admixture --cv plink2_pruned_cbp.bed $K | tee log${K}.out; done

#print cross validation error rates
grep -h CV log*.out > cv_error.log
```

# Conditioning ancestry proportions on clusters in a reference panel -- projection analysis
For this analysis, I wanto to use *Capsella bursa-pastoris* in Eurasia to generate populations in the native range, and assign individuals in the introduced range to those populations.

This requires that the same SNP set be in both datasets. So I will mark SNPs in LD on the whole dataset, and then prune them on the Eurasian samples and NYC samples to generate two sets .bed, .bim, and .fam files

```{bash}
#!/bin/bash
#
#SBATCH --job-name=LDpruneAdmixture
#SBATCH --mem=10G
#SBATCH --time=5:00:00         #realistically, shouldn't take very long
#SBATCH --mail-type=ALL
#SBATCH --mail-user=wils1582@msu.edu
#SBATCH --cpus-per-task=15
#SBATCH --output=/mnt/scratch/wils1582/slurm/slurm-%A_%a.out

### VARIABLES
VCF=/mnt/research/josephslab/Maya/capsella/vcf/CbpCr_Cbp_new/CbpCr_Cbp_new_filtered.vcf.gz
EA_SAMPLES=/mnt/research/josephslab/Maya/capsella/admixture/eurasian_only/eurasian_vcf_cbp.txt
NYC_SAMPLES=/mnt/research/josephslab/Maya/capsella/admixture/nyc_eurasian_conditional/nyc_only.txt
ALL_CBP=/mnt/research/josephslab/Maya/capsella/admixture/all_cbp/vcf_cbp_only.txt
OUTDIR=/mnt/research/josephslab/Maya/capsella/admixture/nyc_eurasian_conditional

# move to directory
cd $OUTDIR

#load modules
module purge
module load PLINK/2.00a_10.2-x86_64
module load ADMIXTURE/1.3.0

#### LD Prune and output BED
# Identify SNPs in LD
# Parameters consider SNPs in 100 snp windows, cutoff is and r^2 correlation of 0.2 between a pair of SNPs, and then calculates again after moving 5 SNPs
plink2 --vcf $VCF \
	     --indep-pairwise 100 5 0.2 \
	     --keep $ALL_CBP \
	--allow-extra-chr \
	--out all_cbp_snps \
	--set-all-var-ids @:#

# Prune LD SNPs
# Variant IDs set to be 'scaffold:position'
# ADMIXTURE requires the input be in PLINK .bed format, so here I create the bed file as well.
  
# Prune Eurasian CBP
plink2 --vcf $VCF \
  --extract all_cbp_snps.prune.in \
  --make-bed \
  --out eurasian_pruned_cbp \
	--allow-extra-chr \
	--keep $EA_SAMPLES \
	--set-all-var-ids @:#
  
# Prune New York/New Jersey CBP
plink2 --vcf $VCF \
  --extract all_cbp_snps.prune.in \
  --make-bed \
  --out nyc_pruned_cbp \
	--allow-extra-chr \
	--keep $NYC_SAMPLES \
	--set-all-var-ids @:#
	
# Check if the .bim files are the same
diff -s eurasian_pruned_cbp.bim nyc_pruned_cbp.bim

# ADMIXTURE only takes chromosomes as numbers so in all the PLINK output files, SCF_ needs to be removed
sed -i 's/^SCF_//g' *.bed
sed -i 's/^SCF_//g' *.bim
sed -i 's/^SCF_//g' *.fam
  
# Run unsupervised ADMIXTURE with K=2
admixture eurasian_pruned_cbp.bed 2

# Use learned allele frequencies as (fixed) input to next step
cp eurasian_pruned_cbp.2.P nyc_pruned_cbp.2.P.in

# Run projection ADMIXTURE with K=2
admixture -P nyc_pruned_cbp.bed 2

### K = 4
# Run unsupervised ADMIXTURE with K=4
admixture eurasian_pruned_cbp.bed 4

# Use learned allele frequencies as (fixed) input to next step
cp eurasian_pruned_cbp.4.P nyc_pruned_cbp.4.P.in

# Run projection ADMIXTURE with K=4
admixture -P nyc_pruned_cbp.bed 4
```

